test/filter-map-builder.test.ts

```diff
@@ -0,0 +1,140 @@
+import { expect } from '@open-wc/testing';
+import { FilterMapBuilder } from '../src/filter-map-builder';
+import { FilterConstraint, FilterMap } from '../src/search-params';
+
+describe('filter map builder', () => {
+  it('initializes with empty filter map', () => {
+    expect(new FilterMapBuilder().build()).to.deep.equal({});
+  });
+
+  it('can add filters', () => {
+    const builder = new FilterMapBuilder();
+    builder.addFilter('foo', 'bar', FilterConstraint.INCLUDE);
+    expect(builder.build()).to.deep.equal({ foo: { bar: 'inc' } });
+
+    builder.addFilter('baz', 'boop', FilterConstraint.EXCLUDE);
+    expect(builder.build()).to.deep.equal({
+      foo: { bar: 'inc' },
+      baz: { boop: 'exc' },
+    });
+
+    builder.addFilter('foo', 'beep', FilterConstraint.GREATER_THAN);
+    expect(builder.build()).to.deep.equal({
+      foo: { bar: 'inc', beep: 'gt' },
+      baz: { boop: 'exc' },
+    });
+  });
+
+  it('can add multiple constraints for one value', () => {
+    const builder = new FilterMapBuilder();
+    builder.addFilter('foo', 'bar', FilterConstraint.INCLUDE);
+    expect(builder.build()).to.deep.equal({ foo: { bar: 'inc' } });
+
+    builder.addFilter('foo', 'bar', FilterConstraint.GREATER_OR_EQUAL);
+    expect(builder.build()).to.deep.equal({
+      foo: { bar: ['inc', 'gte'] },
+    });
+  });
+
+  it('can remove filters', () => {
+    const builder = new FilterMapBuilder();
+    builder.addFilter('foo', 'bar', FilterConstraint.INCLUDE);
+    builder.addFilter('baz', 'boop', FilterConstraint.EXCLUDE);
+    builder.addFilter('foo', 'beep', FilterConstraint.GREATER_THAN);
+    expect(builder.build()).to.deep.equal({
+      foo: { bar: 'inc', beep: 'gt' },
+      baz: { boop: 'exc' },
+    });
+
+    builder.removeFilters('foo', 'bar');
+    expect(builder.build()).to.deep.equal({
+      foo: { beep: 'gt' },
+      baz: { boop: 'exc' },
+    });
+
+    builder.removeFilters('foo', 'beep');
+    expect(builder.build()).to.deep.equal({ baz: { boop: 'exc' } });
+
+    builder.removeFilters('not', 'exist');
+    expect(builder.build()).to.deep.equal({ baz: { boop: 'exc' } });
+
+    builder.removeFilters('baz', 'boop');
+    expect(builder.build()).to.deep.equal({});
+  });
+
+  it('can remove single filters by constraint type', () => {
+    const builder = new FilterMapBuilder();
```

Comment:
this block of code setting up filter state is repeated here and in the `can remove filters` example. since it's not central to the examples demonstrating removing filters, I would consider extracting out to a helper method to focus the remove filter example.